### PR TITLE
Fix Scaffolder Prepare step when templates on disk

### PR DESCRIPTION
**Pull request by benjdlambert**
## Hey, I just made a Pull Request!
- Template entity location should be a directory name and resolve from there
- And actually copy to the path that the templater expects `/template` rather than `/checkout` as there is no need for the `/checkout` folder.

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4516-tlb7t2dncxfwviyjc1skrmnvykby4blr.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6026fb1d65f817cc53ef9d7a

File: .changeset/soft-rings-obey.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@backstage/plugin-scaffolder-backend': patch
+---
+
+Fixed the `prepare` step for when using local templates that were added to the catalog using the `file:` target configuration.
+No more `EPERM: operation not permitted` error messages. ðŸ‘�
```
Comment (freben) [inline .changeset/soft-rings-obey.md:6]: Line noise at end of line? :)

**Review by freben — APPROVED** [review]

File: .changeset/soft-rings-obey.md (hunk 1 above)
Comment (benjdlambert) [reply .changeset/soft-rings-obey.md:6]: Oj. A wild string appeared.

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4516-tlb7t2dncxfwviyjc1skrmnvykby4blr.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6026fb1d65f817cc53ef9d7a

**Pull request merged**